### PR TITLE
Add catch block for failed file promise in `FileRepository`

### PR DESCRIPTION
**Pull request by f1ames**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Add catch block for failed file promise in `FileRepository`.

---

### Additional information

Related to https://github.com/ckeditor/ckeditor5-image/pull/264. In cases where `File` constructor is not supported or file promise fails for any other reason, `loader.file.then` may throw an error if there is no `catch` there (or anywhere in the code).

image

In short, if there is no `loader.file.catch` in the code (some upload adapters may have it, some not), the code before modification threw an error on rejected promise.

### CODE REVIEW

**Comment by jodator** [comment]
@f1ames: Sorry for longer review - at least one test is needed for a change :) Other then that I can see we can close this and ckeditor/ckeditor5-image#264 with this one :)
